src/napari/_vispy/layers/image.py

```diff
@@ -43,19 +50,37 @@ def __init__(
             method='auto',
             texture_format=texture_format,
         )
+        self._tiledimage_node = TiledImageNode(
+            np.array([[0.0]], dtype=np.float32),
+            tile_size=self.MAX_TEXTURE_SIZE_2D,
+            texture_format=texture_format,
+        )
         self._volume_node = VolumeNode(
             np.zeros((1, 1, 1), dtype=np.float32),
             clim=[0, 1],
             texture_format=texture_format,
         )
 
-    def get_node(self, ndisplay: int, dtype: np.dtype | None = None) -> Node:
+    def get_node(
+        self,
+        ndisplay: int,
+        dtype: np.dtype | None = None,
+        shape: tuple | None = None,
+    ) -> Node:
         # Return custom node if we have one.
         if self._custom_node is not None:
             return self._custom_node
 
         # Return Image or Volume node based on 2D or 3D.
-        res = self._image_node if ndisplay == 2 else self._volume_node
+        if ndisplay == 2:
+            if shape is None or not np.any(
+                np.greater(shape, self.MAX_TEXTURE_SIZE_2D)
+            ):
+                res = self._image_node
+            else:
+                res = self._tiledimage_node
+        else:
+            res = self._volume_node
```

Comment:
Here's an equivalent match expression that I find easier to follow than the nested if/elses. What do you think?

```suggestion
        M2D = self.MAX_TEXTURE_SIZE_2D
        match ndisplay, shape:
            # 2D grayscale or RGB w/ any dimension exceeding max texture size
            case 2, (s0, s1, *_) if s0 > M2D or s1 > M2D:
                res = self._tiledimage_node
            # any other 2D
            case 2, _:
                res = self._image_node
            # 3D
            case _:
                res = self._volume_node
```

Reply:
I very rarely use (if ever?) match statements, but I agree that here it makes the logic more readable!